AVInput/AVInput.cpp

```diff
@@ -224,5 +327,923 @@ string AVInput::currentVideoMode(bool &success)
+    }
+
+    if (!parameters.HasLabel("typeOfInput")) {
+        result = false;
+        LOGWARN("please specify type of input HDMI/COMPOSITE");
+    }
+
+    if (result) {
+        int x = 0;
+        int y = 0;
+        int w = 0;
+        int h = 0;
+        int t = 0;
+        string sType;
+
+        try {
+            if (parameters.HasLabel("x")) {
+                x = std::stoi(parameters["x"].String());
+            }
+            if (parameters.HasLabel("y")) {
+                y = std::stoi(parameters["y"].String());
+            }
+            if (parameters.HasLabel("w")) {
+                w = std::stoi(parameters["w"].String());
+            }
+            if (parameters.HasLabel("h")) {
+                h = std::stoi(parameters["h"].String());
+            }
+            if (parameters.HasLabel("typeOfInput")) {
+                sType = parameters["typeOfInput"].String();
+                t = getTypeOfInput (sType);
+            }
+        }
+        catch (...) {
+            LOGWARN("Invalid Arguments");
+            return Core::ERROR_BAD_REQUEST;
+        }
+
+        result = setVideoRectangle(x, y, w, h, t);
+        if (false == result) {
+            LOGWARN("AVInputService::setVideoRectangle Failed");
+            return Core::ERROR_GENERAL;
+        }
+        return Core::ERROR_NONE;
+    }
+    return Core::ERROR_BAD_REQUEST;
+}
+
+bool AVInput::setVideoRectangle(int x, int y, int width, int height, int type)
+{
+    bool ret = true;
+
+    try
+    {
+        if (type == HDMI) {
+            device::HdmiInput::getInstance().scaleVideo(x, y, width, height);
+        }
+        else {
+            device::CompositeInput::getInstance().scaleVideo(x, y, width, height);
+        }
+    }
+    catch (const device::Exception& err) {
+        ret = false;
+    }
+
+    return ret;
+}
+
+uint32_t AVInput::getInputDevicesWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+
+    if (parameters.HasLabel("typeOfInput")) {
+        string sType = parameters["typeOfInput"].String();
+        int iType = 0;
+        try {
+            iType = getTypeOfInput (sType);
+        }catch (...) {
+            LOGWARN("Invalid Arguments");
+            return Core::ERROR_BAD_REQUEST;
+        }
+        response["devices"] = getInputDevices(iType);
+    }
+    else {
+        JsonArray listHdmi = getInputDevices(HDMI);
+        JsonArray listComposite = getInputDevices(COMPOSITE);
+        for (int i = 0; i < listComposite.Length(); i++) {
+            listHdmi.Add(listComposite.Get(i));
+        }
+        response["devices"] = listHdmi;
+    }
+    return Core::ERROR_NONE;
+}
+
+uint32_t AVInput::writeEDIDWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+
+    int deviceId;
+    std::string message;
+
+    if (parameters.HasLabel("deviceId") && parameters.HasLabel("message")) {
+        getNumberParameter("deviceId", deviceId);
```

Comment:
please use `parameters["deviceId"].Number()` if it is number

Reply:
Modified code to use Number()